src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/actions/select/BuckDebugSelectedTestFromEditorAction.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.facebook.buck.intellij.ideabuck.actions.select;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.psi.PsiElement;
+
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.Icon;
+
+/**
+ * Custom Action to right click on a unit test and debug the test from the editor.
+ */
+public final class BuckDebugSelectedTestFromEditorAction extends RunSelectedTestAction {
+
+  BuckDebugSelectedTestFromEditorAction(
+      @Nullable String text,
+      @Nullable String description,
+      @Nullable Icon icon,
+      PsiElement psiElement) {
+    super(text, description, icon);
+    this.element = psiElement;
+  }
+
+  @Override
+  public void actionPerformed(final AnActionEvent event) {
+    setupAndExecuteTestConfigurationFromEditor(event, true);
```

Comment:
Why not add a boolean parameter to `RunSelectedTestAction` and it in `SelectedTestRunLineMarkerContributor`? Creating two classes just because of one boolean seems excessive.